Assets/Scripts/Mecha/Gunner/CannonBehavior.cs

```diff
@@ -43,9 +51,29 @@ void Update ()
             EventManager.onGunnerShot.Invoke(ShotType.Missile);
             // for the scoring script
             EventManager.onGunnerConsumesEnergy.Invoke(MissileEnergyConsumption);
-            Instantiate(MissileShot, Muzzle.position, Muzzle.rotation);
+            projectile = Instantiate(MissileShot, Muzzle.position, Muzzle.rotation);
             mechaController.ConsumeEnergy(MissileEnergyConsumption);
         }
+
+        Ray ray = cam.ScreenPointToRay(Input.mousePosition);
```

Comment:
Instead of doing all of this at each update, put it in a method that is called only when a projectile is fired